nvflare/fuel/hci/tools/admin.py

```diff
@@ -12,90 +12,99 @@
+
+    admin_config = conf.config_data["admin"]
     modules = []
 
-    if args.with_file_transfer:
+    if admin_config["with_file_transfer"]:
         modules.append(
             FileTransferModule(
-                upload_dir=args.upload_dir,
-                download_dir=args.download_dir,
+                upload_dir=admin_config["upload_dir"],
+                download_dir=admin_config["download_dir"],
                 upload_folder_cmd_name=args.upload_folder_cmd_name,
             )
         )
 
-    ca_cert = args.ca_cert
-    client_cert = args.client_cert
-    client_key = args.client_key
+    ca_cert = admin_config["ca_cert"]
+    client_cert = admin_config["client_cert"]
+    client_key = admin_config["client_key"]
 
-    if args.with_ssl:
+    if admin_config["with_ssl"]:
         if len(ca_cert) <= 0:
-            print("missing CA Cert file name")
+            print("missing CA Cert file name field ca_cert in fed_admin configuration")
             return
 
         if len(client_cert) <= 0:
-            print("missing Client Cert file name")
+            print("missing Client Cert file name field client_cert in fed_admin configuration")
             return
 
         if len(client_key) <= 0:
-            print("missing Client Key file name")
+            print("missing Client Key file name field client_key in fed_admin configuration")
             return
     else:
         ca_cert = None
         client_key = None
         client_cert = None
 
     if args.with_debug:
-        print("SSL: {}".format(args.with_ssl))
-        print("User Login: {}".format(args.with_login))
-        print("File Transfer: {}".format(args.with_file_transfer))
-
-        if args.with_file_transfer:
-            print("  Upload Dir: {}".format(args.upload_dir))
-            print("  Download Dir: {}".format(args.download_dir))
+        print("SSL: {}".format(admin_config["with_ssl"]))
+        print("File Transfer: {}".format(admin_config["with_file_transfer"]))
 
-    print("Admin Server: {} on port {}".format(args.host, args.port))
+        if admin_config["with_file_transfer"]:
+            print("  Upload Dir: {}".format(admin_config["upload_dir"]))
+            print("  Download Dir: {}".format(admin_config["download_dir"]))
 
     client = AdminClient(
-        host=args.host,
-        port=args.port,
-        prompt=args.prompt,
+        prompt=admin_config["prompt"],
         cmd_modules=modules,
         ca_cert=ca_cert,
         client_cert=client_cert,
         client_key=client_key,
-        require_login=args.with_login,
-        credential_type=CredentialType.PASSWORD if args.cred_type == "password" else CredentialType.CERT,
+        require_login=admin_config["with_login"],
+        credential_type=CredentialType.PASSWORD if admin_config["cred_type"] == "password" else CredentialType.CERT,
         debug=args.with_debug,
+        overseer_end_point=admin_config["overseer_agent"]["args"]["overseer_end_point"],
```

Comment:
The Admin config already creates the overseer-agent instance. We should pass the overseer-agent to the AdminClient, instead of the overseer_end_point only.